examples/ota-provider-app/ota-provider-common/DefaultUserConsentProvider.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <ota-provider-common/DefaultUserConsentProvider.h>
+
+namespace chip {
+namespace ota {
+
+DefaultUserConsentProvider::DefaultUserConsentProvider()
+{
+    for (uint8_t i = 0; i < kMaxUserConsentEntries; i++)
+    {
+        mUserConsentEntries[i].nodeId = kUndefinedNodeId;
+    }
+}
+
+UserConsentState DefaultUserConsentProvider::GetUserConsentState(NodeId nodeId, EndpointId endpoint, uint32_t currentVersion,
```

Comment:
This API overly constraints the UserConsentProvider to a single fabric: a process running the Matter stack, and serving the needs of multiple fabrics within a given binary process would be unable to work against all of these fabrics unambiguously.

Also, this does not account for different product types (VID/PID). Software versions are ambiguous per version.

Finally, I am not sure what the endpoint helps here, but it's probably fine to keep.

The correct arguments would be:

< PeerId, endpointId, productId, vendorId, softwareVersion, metadata >

PeerId can be extracted from an incoming OTA Query action, to identify the peer.